src/System.Security.Cryptography.Csp/src/System/Security/Cryptography/CapiHelper.cs

```diff
@@ -141,48 +150,51 @@ internal static string GetDefaultProvider(int dwType)
         /// <summary>
         /// Creates a new key container
         /// </summary>
-        private static void CreateCSP(CspParameters parameters, bool randomKeyContainer, ref SafeProvHandle safeProvHandle)
+        private static void CreateCSP(CspParameters parameters, bool randomKeyContainer, out SafeProvHandle safeProvHandle)
         {
-            //don't persist in CSP if this is random key container. 
-            safeProvHandle.PersistKeyInCsp = randomKeyContainer ? false : true;
```

Comment:
We are.  It gets set in OpenCSP.  This one just potentially mutilated the value on an existing SafeHandle before letting it get re-allocated in OpenCSP.